### PR TITLE
2218: Automate execution of integration tests

### PR DESCRIPTION
**Pull request by edvbld**
Hi all,

please review this patch that automates the execution of integration tests _if_ the required properties for the integration test are configured. Prior to this patch one would have to comment out the `@Disabled` notation and then make sure that all properties for all test methods where configured. It was also hard to see all properties required for a given test function to run. With this patch all properties are declared in an annotation and test functions whose properties are configured will be run automatically (no need to comment out any annotation).

I have also renamed these tests to end with the suffix `IntegrationTests.java`. For unit tests we use the suffix `Tests.java` and for tests that require properties to be configured we should use the suffix `IntegrationTests.javaa` (since that is what those tests are).

I also choose a different name for the file containing the properties required for a test to run, the file is now named `test.properties` instead of `manual-test-settings.properties`. The new name is quite a bit shorter and the tests are also no longer manual, they are run automatically as long the required properties for a given test are present.

I also added support for including test properties from another file to allow test properties to be shared among multiple developers. Included files must be specified with an absolute path since it is hard to figure out the current working directory that tests are being run from (it also make the configuration more explicit).

Lastly I have slightly refactored the integration tests so that they all follow a similar pattern. There quite a bit more work that needs to be done here: test properties use different naming schemes (camel case vs dot), not all test properties are prefix with the test suite name (could result in name clashes) and some of the test code could use some tender, love and care. This can be taken care of the later, this patch is already quite big.

Thanks,
Erik

### Testing
- [x] Verified that tests are *not* run when properties are missing
- [x] Verified that tests *are* run when properties are present
- [x] Verified that tests are *not* run when no `test.properties` file is present
- [x] Verified that all JIRA integration tests still pass

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace

### Issue
 * [SKARA-2218](https://bugs.openjdk.org/browse/SKARA-2218): Automate execution of integration tests (**Enhancement** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**) ⚠️ Review applies to [5895bb73](https://git.openjdk.org/skara/pull/1627/files/5895bb7380d5ee1d3c4806c50242b49b8208d647)
 * [Zhao Song](https://openjdk.org/census#zsong) (@zhaosongzs - **Reviewer**) ⚠️ Review applies to [5895bb73](https://git.openjdk.org/skara/pull/1627/files/5895bb7380d5ee1d3c4806c50242b49b8208d647)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/skara.git pull/1627/head:pull/1627` \
`$ git checkout pull/1627`

Update a local copy of the PR: \
`$ git checkout pull/1627` \
`$ git pull https://git.openjdk.org/skara.git pull/1627/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 1627`

View PR using the GUI difftool: \
`$ git pr show -t 1627`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/skara/pull/1627.diff">https://git.openjdk.org/skara/pull/1627.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/skara/pull/1627#issuecomment-2033994529)

### CODE REVIEW

**Review by erikj79 — APPROVED** [review]

**Comment by edvbld** [comment]
Thanks for reviewing @erikj79 ! @zhaosongzs could you also take a look?

**Comment by zhaosongzs** [comment]
> Thanks for reviewing @erikj79 ! @zhaosongzs could you also take a look?

Sure

File: forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java [hunk 1]
```diff
@@ -289,35 +319,43 @@ void testRecentCommitComments() throws IOException {
...
     void testDefaultBranchName() {
-        var gitLabRepo = gitLabHost.repository(settings.getProperty("gitlab.repository")).orElseThrow();
-        assertEquals(settings.getProperty("gitlab.repository.branch"), gitLabRepo.defaultBranchName());
+        var gitLabRepo = gitLabHost.repository(props.get("gitlab.repository")).orElseThrow();
+        assertEquals(props.get("gitlab.repository.branch"), gitLabRepo.defaultBranchName());
     }
 
     @Test
+    @EnabledIfTestProperties({"gitlab.user", "gitlab.pat", "gitlab.uri", "gitlab.group",
+                              "gitlab.user"})
```
Comment (zhaosongzs) [inline forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java:331]: "gitlab.user" is duplicated

File: forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java [hunk 2]
```diff
@@ -27,14 +27,14 @@
 import java.util.Arrays;
 import java.util.Properties;
```
Comment (zhaosongzs) [inline forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java:28]: No longer need "import java.util.Properties"

File: forge/src/test/java/org/openjdk/skara/forge/github/GitHubIntegrationTests.java [hunk 3]
```diff
@@ -24,7 +24,8 @@
 
 import java.time.Duration;
 import java.util.Properties;
```
Comment (zhaosongzs) [inline forge/src/test/java/org/openjdk/skara/forge/github/GitHubIntegrationTests.java:26]: No longer need "import java.util.Properties"

**Review by zhaosongzs — APPROVED** [review]
Looks pretty good and I really like it. Previously, running these tests was inconvenient. Now, this patch has made testing much simpler! Thanks for implementing it!

File: forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java (hunk 1 above)
Comment (edvbld) [reply forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java]: Thanks, fixed in latest commit.

File: forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java (hunk 2 above)
Comment (edvbld) [reply forge/src/test/java/org/openjdk/skara/forge/gitlab/GitLabIntegrationTests.java]: Thanks, fixed in latest!

File: forge/src/test/java/org/openjdk/skara/forge/github/GitHubIntegrationTests.java (hunk 3 above)
Comment (edvbld) [reply forge/src/test/java/org/openjdk/skara/forge/github/GitHubIntegrationTests.java]: Thanks, fixed in latest!

**Comment by edvbld** [comment]
Thanks for reviewing @erikj79 and @zhaosongzs!

/integrate

**Pull request closed without merging**